sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -0,0 +1,410 @@
+   *
+   * The Attestation Signing Keys describe the keys which will be used to sign tokens generated by
+   * the attestation service. All tokens emitted by the attestation service will be signed by one
+   * of the certificates listed in the attestation signing keys.
+   *
+   */
+
+  class AttestationClient final {
+  private:
+    Azure::Core::Url m_endpoint;
+    std::string m_apiVersion;
+    std::shared_ptr<Azure::Core::Http::_internal::HttpPipeline> m_pipeline;
+    AttestationTokenValidationOptions m_tokenValidationOptions;
+
+    mutable std::shared_timed_mutex m_sharedStateLock;
+    mutable std::vector<Models::AttestationSigner> m_attestationSigners;
+
+    void CacheAttestationSigners(Azure::Core::Context const& context) const;
+
+  public:
+    /**
+     * @brief Destructor.
+     *
+     */
+    virtual ~AttestationClient() = default;
+
+    /**
+     * @brief Construct a new Attestation Client object
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential OPTIONAL The authentication method to use (required for TPM attestation).
+     * @param options The options to customize the client behavior.
+     */
+    explicit AttestationClient(
+        std::string const& endpoint,
+        std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+        AttestationClientOptions options = AttestationClientOptions());
+
+    explicit AttestationClient(
+        std::string const& endpoint,
+        AttestationClientOptions options = AttestationClientOptions())
+        : AttestationClient(endpoint, nullptr, options)
+    {
+    }
+
+    /**
+     * @brief Returns the version the client was configured with.
+     *
+     * @returns The API version used when communicating with the attestation service.
+     */
+    std::string ClientVersion() const;
+    /**
+     * @brief Construct a new Key Client object from another key client.
+     *
+     * @param attestationClient An existing attestation client.
+     */
+    explicit AttestationClient(AttestationClient const& attestationClient)
+        : m_endpoint(attestationClient.m_endpoint), m_apiVersion(attestationClient.m_apiVersion),
+          m_pipeline(attestationClient.m_pipeline){};
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * Retrieve the OpenID metadata for this attestation service instance..
+     *
+     * @return an {@link AttestationOpenIdMetadata} containing metadata about the specified service
+     * instance.
+     */
+    Response<Models::AttestationOpenIdMetadata> GetOpenIdMetadata(
+        Azure::Core::Context const& context = Azure::Core::Context::ApplicationContext) const;
+
+    /**
+     * @brief Retrieve the attestation signing certificates for this attestation instance.
+     *
+     * @returns Attestation Metadata.
+     */
+    Response<std::vector<Models::AttestationSigner>> GetAttestationSigningCertificates(
```

Comment:
might want to wrap the vector inside a `AttestationSigningCertificageResult` struct as a way to support returning more than just the vector in the future.